### PR TITLE
test

### PR DESCRIPTION
**Pull request by bouvyd**
checking if resetting lazy props for multi company on the environment does not create side effects

### CODE REVIEW

File: odoo/api.py [hunk 1]
```diff
@@ -611,6 +611,9 @@ def clear(self):
         """ Clear all record caches, and discard all fields to recompute.
             This may be useful when recovering from a failed ORM operation.
         """
+        # Since the company/companies lazy properties can contain data that is no longer valid,
+        # (e.g. company created during a test or a failed ORM operation), reset these lazy properties
+        lazy_property.reset_all(self.all)
```
Comment (rco-odoo) [inline odoo/api.py]: This is not correct: the lazy properties are not on `self.all`, but on all environments in `list(self.all)`.

**Review by rco-odoo — COMMENTED** [review]
I was wondering: changing the current user's company/companies should also reset those lazy properties on the current request's environments.  It's not only when you rollback, in fact.

File: odoo/api.py (hunk 1 above)
Comment (rco-odoo) [reply odoo/api.py]:
```suggestion
        for env in list(self.all):
            lazy_property.reset_all(env)
```

**Comment by dbo-odoo** [comment]
@rco-odoo yes, i would tend to agree. Should probably happen on write of `company_id`/`company_ids` on `res.users`, perhaps on write of `active` of `res.company` as well, unlink and create of `res.company` (esp. since create of a company will add it in the companies of the current user IIRC). More stuff perhaps.

**Review by rousseldenis — APPROVED** [review]

**Review by rco-odoo — COMMENTED** [review]
Looks good to me.
But it needs some test :wink:

**Comment by pedrobaeza** [comment]
What's the status of this?

**Comment by dbo-odoo** [comment]
closing this because i'm leaving; if someone wants to pick-it up feel free to fork and re-PR

**Pull request closed without merging**